sympy/series/limits.py

```diff
@@ -109,6 +109,22 @@ def __new__(cls, e, z, z0, dir="+"):
         obj._args = (e, z, z0, dir)
         return obj
 
+
+    @property
+    def free_symbols(self):
+        e = self.args[0]
+        # e and z are same then none is free_symbols
+        if e is self.args[1]:
+            isyms = set()
+        else:
+            isyms = e.free_symbols
+            # take out z from isyms
+            if self.args[1] in isyms:
+                isyms.remove(self.args[1])
```

Comment:
Would it be possible to use ``discard`` without checking the presence of ``self.args[1]``?